### PR TITLE
Jupyter Utilities

### PR DESCRIPTION
**Pull request by ericspod**
### Description
This includes some utilities to help use MONAI in Jupyter. I will have an example notebook shortly for the tutorials repo. One issue is that the type checking was difficult to fully include and make work with Matplotlib so some values are untyped, and `optional_import` cause circular dependency issues in the utility file. 

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/handlers/metric_logger.py [hunk 1]
```diff
@@ -21,12 +23,43 @@
     Engine, _ = optional_import("ignite.engine", "0.4.4", exact_version, "Engine")
 
 
+def _get_loss_from_output(output):
+    return output["loss"].item()
```
Comment (rijobro) [inline monai/handlers/metric_logger.py:27]: Could be:
```suggestion
def _get_loss_from_output(output, loss_key = "loss"):
    return output[loss_key].item()
```

File: monai/handlers/metric_logger.py [hunk 2]
```diff
@@ -21,12 +23,43 @@
...
+
+    Args:
+        loss_transform: Converts the `output` value from the trainer's state into a loss value
+        metric_transform: Converts the metric value coming from the trainer/evaluator's state into a storable value
+        evaluator: Optional evaluator to consume metric results from at the end of its evaluation run
+    """
+
+    def __init__(
+        self,
+        loss_transform: Callable = _get_loss_from_output,
```
Comment (rijobro) [inline monai/handlers/metric_logger.py:50]: Might be overkill but could be:
```suggestion
        loss_transform: Callable = partial(_get_loss_from_output, loss_key=MetricLoggerKeys.LOSS),
```

File: monai/utils/jupyter_utils.py [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+                    imagemap[f"{label}_{i}"] = im
+
+    axes = plot_metric_images(fig, title, graphmap, imagemap, yscale, avg_keys, window_fraction)
+
+    axes[0].axhline(logger.loss[-1][1], c="k", ls=":")  # draw dotted horizontal line at last loss value
+
+    return fig, axes
+
+
+def _get_loss_from_output(output: Union[Dict[str, torch.Tensor], torch.Tensor]) -> float:
```
Comment (rijobro) [inline monai/utils/jupyter_utils.py:232]: Could you call the other implementation to reduce duplication?

**Comment by rijobro** [comment]
Looks good. I really like plotting losses with live updates, which makes my tutorials more bloated than necessary. Would be great if that functionality could be incorporated somehow (see use of `plot_range` in https://app.reviewnb.com/Project-MONAI/tutorials/pull/129/).

**Review by rijobro — APPROVED** [review]

**Comment by ericspod** [comment]
`plot_range` and the other plotting functions should be consolidated in one place, I have other plot functions to add later on.

File: monai/utils/jupyter_utils.py [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+    ax,
+    title: str,
+    graphmap: Dict[str, Union[List[float], Tuple[List[float], List[float]]]],
+    yscale: str = "log",
+    avg_keys: Tuple[str] = (LOSS_NAME,),
+    window_fraction: int = 20,
+):
+    """
+    Plot metrics on a single graph with running averages plotted for selected keys. The values in `graphmap`
+    should be lists of (timepoint, value) pairs as stored in MetricLogger objects.
```
Comment (wyli) [inline monai/utils/jupyter_utils.py:69]: could you create a few tests for the plotting method and MetricLogger? came across this util not sure how reliable it is https://github.com/matplotlib/matplotlib/blob/6fe34e72d2b7daf8586a08a8611cc73b3c0ad60c/lib/matplotlib/testing/decorators.py#L393

Comment (wyli) [reply monai/utils/jupyter_utils.py:69]: I forgot to request changes for unit tests, could you please have a follow-up PR, thank you! @ericspod

**Comment by wyli** [comment]
without tests it's difficult for the other maintainers to understand the intended behaviour and ensure the quality...
as we have a tight schedule for v0.5, If we don't have proper tests for this PR within a week, I'll try to exclude it from the upcoming release to avoid any issues

**Comment by wyli** [comment]
followup testing addressed by #1826

**Pull request merged**